network/src/commonMain/kotlin/com/wire/kalium/network/api/notification/NotificationApiImpl.kt

```diff
@@ -47,30 +47,18 @@ class NotificationApiImpl(private val httpClient: HttpClient, private val server
         queryClient: String,
         querySince: String?
     ): NetworkResponse<NotificationResponse> {
-        return try {
+        return wrapKaliumResponse({
+            if (it.status.value != 404) null
+            else {
+                val body = it.body<NotificationResponse>().copy(isMissingNotifications = true)
+                NetworkResponse.Success(body, it)
+            }
```

Comment:
Yep. `404` is considered a "Success" when fetching notifications. I didn't change this behaviour.

This means that the backend is returning the first page of pending notifications, but is notifying us that the specified `querySync` no longer exists, so some notifications could be missing (full sync needed).